### PR TITLE
feat: legal notice on wallet connect

### PR DESCRIPTION
**Pull request by filipviz**
If a user attempts to connect their wallet, show them a legal notice. Once they accept, they can connect as normal.

If a user accepts: `localStorage.setItem('hasAcceptedLegalNotice', 'true')`.

If this is true, they won't be prompted again.

### CODE REVIEW

**Comment by filipviz** [comment]
image
image

File: src/components/Navbar/components/Wallet/WalletButton.tsx [hunk 1]
```diff
@@ -2,8 +2,10 @@ import { WarningOutlined } from '@ant-design/icons'
 import { Trans } from '@lingui/macro'
 import { Button } from 'antd'
 import { useWallet } from 'hooks/Wallet'
+import LegalNoticeModal from 'components/modals/LegalNoticeModal'
```
Comment (wraeth-eth) [inline src/components/Navbar/components/Wallet/WalletButton.tsx:5]: I wonder if this behaviour should be a wallet signing request instead. NAL, but I am not sure how usable this is as a legal notice. That said, im unsure how usable the wallet signing is either, but we would have a record to point to at very least 🤷‍♂️

File: src/components/Navbar/components/Wallet/WalletButton.tsx [hunk 2]
```diff
@@ -14,6 +16,41 @@ export default function WalletButton() {
     changeNetworks,
   } = useWallet()
 
+  const [legalNoticeAccepted, setLegalNoticeAccepted] = useState<boolean>(
+    localStorage.getItem('hasAcceptedLegalNotice') === 'true',
```
[the remark this answers is not in the document]
Comment (filipviz) [reply src/components/Navbar/components/Wallet/WalletButton.tsx:20]: Works fine on the preview. Would it be different in prod?

[the remark this answers is not in the document]
Comment (filipviz) [reply src/components/Navbar/components/Wallet/WalletButton.tsx]: Makes sense @wraeth-eth @peripheralist. Should be fixed in [0274a25](https://github.com/jbx-protocol/juice-interface/pull/3931/commits/0274a25f0889aaf6f1bc87a69bced8354f138efe)

**Comment by tomquirk** [comment]
@filipviz looking at this now 🙏 . Gonna push a `main` rebase

**Review by tomquirk — APPROVED** [review]
@filipviz approving and merging (note: also included walletconnect overlay fix in this pr)

**Pull request merged**